bindings/python/setup.py

```diff
@@ -147,15 +150,17 @@ def append_arrow_flags(ext):
         pa.create_library_symlinks()
 
     if os.name == "posix":
-        ext.extra_compile_args.append("-std=c++11")
+        ext.extra_compile_args.append("-std=c++17")
 
     # Arrow's manylinux{2010, 2014} binaries are built with gcc < 4.8 which predates CXX11 ABI
     # - https://uwekorn.com/2019/09/15/how-we-build-apache-arrows-manylinux-wheels.html
     # - https://arrow.apache.org/docs/python/extending.html#example
-    if "std=" not in os.environ.get("CXXFLAGS", ""):
-        ext.extra_compile_args.append("-std=c++11")
+    if os.name != "nt" and "std=" not in os.environ.get("CXXFLAGS", ""):
         ext.extra_compile_args.append("-D_GLIBCXX_USE_CXX11_ABI=0")
 
+    if os.name == "nt":
+        ext.extra_compile_args.append("/std:c++17")
```

Comment:
Could we move this up so that it's next to the `if os.name == "posix":`?

Reply:
I grouped things a bit better now